src/cupbearer/detectors/statistical/mahalanobis_detector.py

```diff
@@ -6,11 +6,12 @@
 )
 
 
-def _pinv(C, rcond):
+def _pinv(C, rcond, dtype=torch.float64):
     # Workaround for pinv not being supported on MPS
+    og_dtype = C.dtype
     if C.is_mps:
-        return torch.linalg.pinv(C.cpu(), rcond=rcond, hermitian=True).to(C.device)
-    return torch.linalg.pinv(C, rcond=rcond, hermitian=True)
+        return torch.linalg.pinv(C.cpu().to(dtype), rcond=rcond, hermitian=True).to(og_dtype).to(C.device)
```

Comment:
```suggestion
        return torch.linalg.pinv(C.cpu().to(dtype), rcond=rcond, hermitian=True).to(C.dtype).to(C.device)
```
Nit: Personally I find C.dtype to be clearer with what it means. But, is fine as is as well.